forms/AbiDecodeForm.vue

```diff
@@ -0,0 +1,318 @@
+import { ETHEREUM_TYPES } from '@/enums'
+import {
+  CheckboxField,
+  InputField,
+  RadioButtonField,
+  SelectField,
+  TextareaField,
+} from '@/fields'
+import { ErrorHandler, checkIsBigInt, hex, required } from '@/helpers'
+import { type ArrayElement, type FieldOption } from '@/types'
+import { guessAbiEncodedData, guessFragment } from '@openchainxyz/abi-guesser'
+import { AbiCoder, FunctionFragment } from 'ethers'
+import { debounce } from 'lodash-es'
+import { v4 as uuidv4 } from 'uuid'
+import { computed, reactive, ref, watch } from 'vue'
+import { i18n } from '~/plugins/localization'
+
+type DecodedData = {
+  types: string[]
+  values: unknown[]
+}
+
+type FuncArg = {
+  id: string
+  type: string
+  value: string
+}
+
+enum DECODE_MODES {
+  auto = 'auto',
+  manual = 'manual',
+}
+
+defineProps<{
+  title: string
+}>()
+
+const { t } = i18n.global
+
+const errorMessage = ref('')
+const isDecoding = ref(false)
+
+const addArg = () => form.args.push({ id: uuidv4(), type: '', value: '' })
+const removeArg = (id: FuncArg['id']) => {
+  form.args = form.args.filter(arg => arg.id !== id)
+}
+
+const decodeModeOptions = computed<FieldOption[]>(() => [
+  {
+    value: DECODE_MODES.auto,
+    title: t('abi-decode-form.decode-mode-option-title--auto'),
+  },
+  {
+    value: DECODE_MODES.manual,
+    title: t('abi-decode-form.decode-mode-option-title--manual'),
+  },
+])
+
+const form = reactive({
+  abiEncoding: '',
+  hasFuncSelector: false,
+  decodeMode: decodeModeOptions.value[0].value,
+  args: [] as FuncArg[],
+})
+const rules = computed(() => ({
+  abiEncoding: { required, hex },
+  decodeMode: { required },
+  args: {
+    ...form.args.reduce(
+      (acc, _, idx) => ({
+        ...acc,
+        [idx]: { type: { required } },
+      }),
+      {},
+    ),
+  },
+}))
+
+const { getFieldErrorMessage, isFieldsValid, isFormValid, touchField } =
+  useFormValidation(form, rules)
+
+const fetchFuncSignature = async (selector: string): Promise<string> => {
+  const url = 'https://api.openchain.xyz/signature-database/v1/lookup'
+  const query = `function=${selector}`
+
+  const response = await (await fetch(`${url}?${query}`)).json()
+  if (!response.ok) throw new Error(response.error)
+
+  return response.result.function[selector]?.[0]?.name || ''
+}
+
+const decode = async (): Promise<DecodedData> => {
+  isDecoding.value = true
+
+  let types: DecodedData['types'] = []
+  let values: DecodedData['values'] = []
+
+  try {
+    switch (true) {
+      case form.decodeMode === DECODE_MODES.auto && form.hasFuncSelector: {
+        const funcSelector = form.abiEncoding.substring(0, 10)
+        const funcData = '0x' + form.abiEncoding.substring(10)
+
+        const funcSignature = await fetchFuncSignature(funcSelector)
+
+        const funcFragment = funcSignature
+          ? FunctionFragment.from(funcSignature)
+          : guessFragment(form.abiEncoding)
+        if (!funcFragment) throw new Error('failed guess fragment')
```

Comment:
1. it's not localised
2. I don't know how "guessing fragment" works. Do you process this error somewhere else? If not - this description is not enough.
3. why there is no default statement?

Reply:
TODO: errors localization